sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/ingestion/http/IngestionHttp.java

```diff
@@ -82,6 +82,12 @@
     private static final int READ_TIMEOUT = 20000;
 
     /**
+     * Regex pattern for application secret in header.
```

Comment:
Overkill to use regex like this but ok...

However appSecret can be something else than UUID soon (for codepush migration or something like that), that's why we removed the client side check earlier.

Your code will not obfuscate an appSecret that does not match UUID format. Why not using a simple substring based code that does not assume anything about format (except handling short length edge case)?